Makefile.common

```diff
@@ -33,7 +35,7 @@ endif
 
 # Check that rustc is the right nightly - warn not error if wrong, sometimes useful to test others
 RUSTC_VERSION_STRING := rustc 1.16.0-nightly (83c2d9523 2017-01-24)
-ifneq ($(shell rustc --version),$(RUSTC_VERSION_STRING))
+ifneq ($(shell RUSTUP_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustc --version),$(RUSTC_VERSION_STRING))
     $(warning Unexpected rustc version. Expected $(RUSTC_VERSION_STRING) got $(shell rustc --version))
     $(warning You may experience unexpected compilation warnings or errors)
     $(warning To fix, run `rustup override set nightly-2017-01-25`)
```

Comment:
Should we change this line to just suggest installing `rustup` instead of suggesting running override?

Also, if I've done an override _and_ this env variable is set - who wins? Importantly, if the env variable wins, we need to have an escape hatch to let people try building with different versions briefly.